internal/locald/sandboxmanager/grpc_server.go

```diff
@@ -262,3 +266,23 @@ func (s *grpcServer) sbStatuses() []*commonapi.SandboxStatus {
 	}
 	return res
 }
+
+func (s *grpcServer) stop() {
+	var wg sync.WaitGroup
+
+	// stop all sbx monitors
+	s.sbMu.Lock()
+	for _, sbMon := range s.sbMonitors {
+		wg.Add(1)
+		// overwrite the delete function
+		sbMon.delFn = func() {
+			defer wg.Done()
+		}
+		// stop the sbx monitor
+		sbMon.stop()
```

Comment:
I find this flow a bit hard to follow and would suggest
1. Adding the idea of a `doneAck` to `sbMon`
2. having `sbMon.stop` wait for the `<-doneAck`
3. keeping the `delFn` as is

Reply:
That would cause a deadlock because the `sbMon.delFn` locks `s.sbMu` too:

https://github.com/signadot/cli/blob/904e7b059b749c2b422f1d181dd2911c159e6c3a/internal/locald/sandboxmanager/grpc_server.go#L158-L160

The other idea I thought of was waiting until `len(s.sbMonitors) == 0`, but I liked more what I ended up implementing.
But I'm open to changing it if you think there's a better approach.

Reply:
I guess it's just inherently messy any which way.   So, your preference.